llpc/patch/llpcPatchIntrinsicSimplify.cpp

```diff
@@ -0,0 +1,416 @@
+ *
+ *  Copyright (c) 2017-2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcPatchIntrinsicSimplify.cpp
+ * @brief LLPC source file: contains implementation of class Llpc::PatchIntrinsicSimplify.
+ ***********************************************************************************************************************
+ */
+#define DEBUG_TYPE "llpc-patch-intrinsic-simplify"
+
+#if defined(_MSC_VER)
+// Provide M_PI.
+#define _USE_MATH_DEFINES
+#endif
+
+#include <math.h>
+
+#include "llvm/InitializePasses.h"
+#include "llvm/Analysis/ScalarEvolution.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
+
+#include "llpcPatchIntrinsicSimplify.h"
+#include "llpcTargetInfo.h"
+
+using namespace llvm;
+using namespace Llpc;
+
+namespace Llpc
+{
+
+// =====================================================================================================================
+// Initializes static members.
+char PatchIntrinsicSimplify::ID = 0;
+
+// =====================================================================================================================
+// Pass creator, creates the LLVM pass intrinsic simplifcations.
+FunctionPass* CreatePatchIntrinsicSimplify()
+{
+    return new PatchIntrinsicSimplify();
+}
+
+// =====================================================================================================================
+// Constructor.
+PatchIntrinsicSimplify::PatchIntrinsicSimplify()
+    :
+    FunctionPass(ID),
+    m_pScalarEvolution(nullptr),
```

Comment:
Why not assign these in the class definition? I usually start reading class code in the header and then immediately wonder what happens with the uninitialized members. If they are initialized in the header, this worry immediately disappears.